pkg/controller/cloneset/cloneset_controller.go

```diff
@@ -498,3 +505,27 @@ func (r *ReconcileCloneSet) truncateHistory(
 	}
 	return nil
 }
+
+func (r *ReconcileCloneSet) claimPods(instance *appsv1alpha1.CloneSet, pods []*v1.Pod) ([]*v1.Pod, error) {
+	manager, err := refmanager.New(r, instance.Spec.Selector, instance, r.scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	selected := make([]metav1.Object, len(pods))
+	for i, pod := range pods {
+		selected[i] = pod.DeepCopy()
```

Comment:
There is no need to deep copy. The Pods list from controller-runtime have already been deep copied.
https://github.com/openkruise/kruise/blob/master/vendor/sigs.k8s.io/controller-runtime/pkg/cache/internal/cache_reader.go#L138